addons/survey/static/src/js/survey_session_chart.js

```diff
@@ -171,6 +171,40 @@ publicWidget.registry.SurveySessionChart = publicWidget.Widget.extend({
                     }
                 }
             },
+            plugins: [{
+                beforeInit: function (chart) {
+                    const nbrCol = chart.data.labels.length;
+                    let charPerLine = 25;
+                    const maxWordLength = chart.data.labels.join(' ').split(' ').sort((a, b) => b.length - a.length)[0].length;
+                    if (nbrCol > 4) {
+                        charPerLine = nbrCol > 8 ? 7 : nbrCol > 6 ? 10 : 15;
+                        chart.options.scales.xAxes[0].ticks.fontSize = maxWordLength > charPerLine ? 15 : chart.width * (nbrCol) / 100;
+                    }
+                    chart.data.labels.forEach(function (value, index, array) {
+                        let test = value.split(" ");
+                        var a = [];
+                        let tmp = [];
+                        let actualLength = 0;
+                        let j = 0;
+                        for (let i = 0; i < test.length; i++) {
+                            tmp.push(test[i]);
+                            actualLength += tmp[j].length
+                            if (
+                                actualLength < charPerLine - tmp.length + 1 && i+1 < test.length &&
+                                tmp.join(' ').length + test[i+1].length < charPerLine
+                            ) {
+                                j++;
+                                continue;
+                            }
+                            let newPart = tmp.join(' ');
+                            a.push(newPart);
+                            tmp = [];
+                            actualLength = j = 0;
+                        }
+                        array[index] = a;
+                    });
+                },
+            }],
```

Comment:
Hello @jeh-odoo 

This is going to sound a bit harsh, but this PR has a lot of issues even though it's pretty small:
- Commit message has nothing to do with changes you make
- No code comment to explain this (and considering the complexity it sure needs one)
- Strange mix of var/let/const, where I'm pretty sure a lot of "let" could be "const"
  (Either you put "var" everywhere, the old school way, either you pay attention to where you need const and where you need let)
- Must improve variable naming, e.g: "a", "test', "tmp" and "j" really hurt code comprehension
- There are "magic numbers" everywhere, could use some well named constants to ease reading
- `if (actualLength < charPerLine - tmp.length + 1 && i+1 < test.length &&  tmp.join(' ').length + test[i+1].length < charPerLine)`
  ...what? Could use well named condition-variable and explanations

I think you can do a lot better than that :) 

Side note: this commit should target 14.0 as saas-13.3 is dead.